x/tokenomics/keeper/update_relay_mining_difficulty.go

```diff
@@ -156,6 +166,22 @@ func computeEma(alpha float64, prevEma, currValue uint64) uint64 {
 	return uint64(alpha*float64(currValue) + (1-alpha)*float64(prevEma))
 }
 
+// getNumLeadingZeroBitsFromHash returns the number of leading zero bits for the target hash.
+func getNumLeadingZeroBitsFromHash(targetHash []byte) int {
```

Comment:
```suggestion
// relayMiningTargetHashToDifficulty returns relay mining difficulty based on the hash.
// This currently implies the number of leading zero bits but may changed in the future.
// TODO_MAINNET: Determine if we should launch with a more adaptive difficulty or stick to
// leading zeroes.
func relayMiningTargetHashToDifficulty(targetHash []byte) int {
```

See this conversation for reference: https://bitcoin.stackexchange.com/questions/121920/is-it-always-possible-to-find-a-number-whose-hash-starts-with-a-certain-number-o

<img width="769" alt="Screenshot 2024-06-21 at 6 25 14 PM" src="https://github.com/pokt-network/poktroll/assets/1892194/6ad22a1f-ff74-47b9-996d-2967b2270205">

Reply:
Consider adding a comment, TODO or renaming it elsewhere too.